### PR TITLE
Generalizing the weather processing system for atoms.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- `SSweather_atoms` handles processing atoms that care about weather (humans and cleanable decals currently).
- Most cleanable decals will slowly fade away while rained on. 
- Weather ambience for mobs is handled by `handle_environment()` while exposure effects are handled by `process_weather()`.
- `get_affecting_weather()` and `is_outside()` have been generalized to `/atom`.

### CODE REVIEW

File: code/modules/mob/living/life.dm [hunk 1]
```diff
@@ -291,18 +277,22 @@
...
-/mob/living/proc/handle_weather_effects(obj/abstract/weather_system/weather)
-	// Handle physical effects of weather.
-	if(!istype(weather))
-		return
-	var/decl/state/weather/weather_state = weather.weather_system.current_state
-	if(istype(weather_state))
-		weather_state.handle_exposure(src, get_weather_exposure(weather), weather)
+/mob/living/process_weather(obj/abstract/weather_system/weather, decl/state/weather/weather_state)
+	// Handle physical effects of weather. Ambience is handled in handle_environment
+	// as clients should get wind noise etc. regardless of weather sensitivity.
```
Comment (out-of-phaze) [inline code/modules/mob/living/life.dm]: i don't think this is correct, i think `WEATHER_IGNORE` should block ambience, while `WEATHER_EXPOSED`, `WEATHER_ROOFED`, and `WEATHER_PROTECTION` should allow it

Comment (MistakeNot4892) [reply code/modules/mob/living/life.dm]: Why would WEATHER_IGNORE block ambience? Ambience is stuff like rain drumming on the roof or wind noises.

Comment (MistakeNot4892) [reply code/modules/mob/living/life.dm]: Nvm I'm stupid and forgot my own code, WEATHER_IGNORE skips ambience anyway. Dunno if I like that on reflection, it'd be nice to hear rain from inside a locker or tent or such.

Comment (MistakeNot4892) [reply code/modules/mob/living/life.dm]: Updated the comment but I think the structure is sound. Mobs without clients don't need to do any weather ambience processing and the other proc has a client check.

Comment (out-of-phaze) [reply code/modules/mob/living/life.dm]: I'm pretty sure `WEATHER_IGNORE` is specifically for "you are two z-levels down and there is no way the weather could reach you," IIRC? Like how we wouldn't want rain sounds to be audible from within the deep caverns

Comment (MistakeNot4892) [reply code/modules/mob/living/life.dm]: I'm not touching that logic so I'm a bit confused by what this comment thread wants me to do.

Comment (out-of-phaze) [reply code/modules/mob/living/life.dm]: was just replying to `Dunno if I like that on reflection, it'd be nice to hear rain from inside a locker or tent or such.`, no request for changes. as evidenced by how there were not changes requested

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
Has an extra commit that shouldn't be in there.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**